### PR TITLE
Revert "Revert "Add PDL pre-processing pass to compiler" (#16968)"

### PR DESCRIPTION
**Pull request by nirvedhmeshram**
Adding a fix to pass the failing tf tests
This reverts commit https://github.com/openxla/iree/commit/72681a78e82bcc5b2e229f9ef0ea63765b6f595a.

### CODE REVIEW

File: compiler/src/iree/compiler/Preprocessing/Passes.cpp [hunk 1]
```diff
@@ -90,9 +90,9 @@ void buildPreprocessingPassPipeline(
         preprocessingOptions.preprocessingPDLSpecFilename;
     passManager.addPass(
         Preprocessing::createApplyPDLPatterns(applyPDLPatternsOptions));
+    passManager.addPass(createCanonicalizerPass());
+    passManager.addPass(createCSEPass());
   }
-  passManager.addPass(createCanonicalizerPass());
-  passManager.addPass(createCSEPass());
```
Comment (ScottTodd) [inline compiler/src/iree/compiler/Preprocessing/Passes.cpp]: Can you elaborate on how this fixed the failing test? These passes shouldn't really be loading bearing (either including or excluding them)

Comment (nirvedhmeshram) [reply compiler/src/iree/compiler/Preprocessing/Passes.cpp]: I added these passes to simplify lit tests that I am adding in the PR that looks at the iree-compile IR with the --compile-to=preprocessing stage. It is not load-bearing for the mlp-plugin work this PR is enabling. So why that crashes the tf test is something the engineers developing/maintaining that pipeline should investigate.

Comment (nirvedhmeshram) [reply compiler/src/iree/compiler/Preprocessing/Passes.cpp]: oh and to answer how this fixes it, I am only running the clean up passes when I am running the pdl pass now, so it doesnt change the flow for other cases this way.

Comment (ScottTodd) [reply compiler/src/iree/compiler/Preprocessing/Passes.cpp]: Weird that the compilation flow is brittle there. Thanks for the fix.

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**